### PR TITLE
Some code clean ups.

### PR DESCRIPTION
**Pull request by dromato**
Nothing that could change logic or application flow, just minor refactors to be consistent with good practices and clean code.
To sum up changes:
- Simplified some conditions
- Changed small l to L in long number, as l tends to look like 1 and might confuse
- Removed "return" statements where those are not necessary (last instruction in function)
- Inlined returns where there were no need for creating new variable (easier to read)
- Deleted unnecessary colons

### CODE REVIEW

File: src/main/java/rx/internal/operators/OnSubscribeRedo.java [hunk 1]
```diff
@@ -195,7 +195,7 @@ public void call(final Subscriber<? super T> child) {
         final AtomicBoolean resumeBoundary = new AtomicBoolean(true);
         
         // incremented when requests are made, decremented when requests are fulfilled
-        final AtomicLong consumerCapacity = new AtomicLong(0l);
+        final AtomicLong consumerCapacity = new AtomicLong(0L);
```
Comment (akarnokd) [inline src/main/java/rx/internal/operators/OnSubscribeRedo.java]: Unnecessary default value.

Comment (dromato) [inline src/main/java/rx/internal/operators/OnSubscribeRedo.java]: Fixed.

File: src/main/java/rx/internal/operators/OperatorZip.java [hunk 2]
```diff
@@ -112,8 +112,7 @@ public OperatorZip(Func9 f) {
         final Zip<R> zipper = new Zip<R>(child, zipFunction);
         final ZipProducer<R> producer = new ZipProducer<R>(zipper);
         child.setProducer(producer);
-        final ZipSubscriber subscriber = new ZipSubscriber(child, zipper, producer);
-        return subscriber;
+        return new ZipSubscriber(child, zipper, producer);
```
Comment (akarnokd) [inline src/main/java/rx/internal/operators/OperatorZip.java]: This will conflict with a zip patch. Could you undo this?

Comment (dromato) [inline src/main/java/rx/internal/operators/OperatorZip.java]: Done :).

**Comment by akarnokd** [comment]
Could you also squash the commits?

**Comment by dromato** [comment]
Squashed :)

**Comment by akarnokd** [comment]
:+1:

File: src/test/java/rx/internal/util/IndexedRingBufferTest.java [hunk 3]
```diff
@@ -191,11 +191,7 @@ public Boolean call(String t1) {
             @Override
             public Boolean call(String t1) {
                 list.add(t1);
-                if (i++ == 2) {
-                    return false;
-                } else {
-                    return true;
-                }
+                return i++ != 2;
```
Comment (artem-zinnatullin) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: Programming on side effects :crying_cat_face:

**Comment by artem-zinnatullin** [comment]
:+1:

File: src/test/java/rx/internal/util/IndexedRingBufferTest.java (hunk 3 above)
Comment (dromato) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: Well I didn't want to change any existing logic there... 
I don't know code base well enough to feel comfortable enough.
I could take that "i++" one separate line higher.

Comment (artem-zinnatullin) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: >I could take that "i++" one separate line higher.

That's why I don't like such code, it's non-obvious (but short), in fact, "post `++`" will be applied after the condition check, so correct transformation would be:

```java
final boolean result = i != 2;
i++;
return result;
``` 

At least, we don't have undefined behavior here, so, you can leave it as is, but personally, I'd rewrite it.

Comment (dromato) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]:
```java
i++;
return i != 3;
```

Should also be fine, right?
We increment both sides by 1.

Comment (artem-zinnatullin) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: Yep.

Comment (dromato) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: Changed.

Comment (artem-zinnatullin) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: Thanks!

Comment (dromato) [inline src/test/java/rx/internal/util/IndexedRingBufferTest.java]: My pleasure :) :smiley:

**Comment by abersnaze** [comment]
looks good.

**Pull request merged**